packages/rule-sri/src/sri.ts

```diff
@@ -0,0 +1,305 @@
+     */
+    private async isEligibleForIntegrityValidation(evt: FetchEnd): Promise<boolean> {
+        debug('Is eligible for integrity validation?');
+
+        const { element, resource } = evt;
+        const resourceOrigin: string = new URL(resource).origin;
+
+        // CORS validation only applies to scripts, styles are OK
+        if (element.nodeName !== 'SCRIPT') {
+            return true;
+        }
+
+        if (this.origin === resourceOrigin) {
+            return true;
+        }
+
+        // cross-origin scripts need to be loaded with a valid "crossorigin" attribute (ie.: anonymous or use-credentials)
+        const crossorigin = element.getAttribute('crossorigin');
+
+        if (!crossorigin) {
+            await this.context.report(resource, element, `Cross-origin scripts need a "crossorigin" attribute to be elegible for integrity validation`);
+
+            return false;
+        }
+
+        const validCrossorigin = crossorigin === 'anonymous' || crossorigin === 'use-credentials';
+
+        if (!validCrossorigin) {
+            await this.context.report(resource, element, `Attribute "crossorigin" doesn't have a valid value, should "anonymous" or "use-credentials": crossorigin="${crossorigin}"`);
+        }
+
+        return validCrossorigin;
+    }
+
+    /** Checks if the element that triggered the download has the `integrity` attribute. */
+    private async hasIntegrityAttribute(evt: FetchEnd): Promise<boolean> {
+        debug('has integrity attribute?');
+        const { element, resource } = evt;
+        const integrity = element.getAttribute('integrity');
+
+        if (!integrity) {
+            await this.context.report(resource, element, `Resource ${resource} requested without the "integrity" attribute`);
+        }
+
+        return !!integrity;
+    }
+
+    /**
+     * Checks if the format of the `integrity` attribute is valid and if the used hash meets
+     * the baseline (by default sha-384). In the case of multiple algorithms used, the
+     * one with the highest priority is the used one to validate. E.g.:
+     *
+     * * `<script src="https://example.com/example-framework.js"
+     *   integrity="sha384-Li9vy3DqF8tnTXuiaAJuML3ky+er10rcgNR/VqsVpcw+ThHmYcwiB1pbOxEbzJr7"
+     *   crossorigin="anonymous"></script>`
+     * * `<script src="https://example.com/example-framework.js"
+     *   integrity="sha384-Li9vy3DqF8tnTXuiaAJuML3ky+er10rcgNR/VqsVpcw+ThHmYcwiB1pbOxEbzJr7
+     *              sha384-+/M6kredJcxdsqkczBUjMLvqyHb1K/JThDXWsBVxMEeZHEaMKEOEct339VItX1zB"
+     *   crossorigin="anonymous"></script>`
+     *
+     * https://w3c.github.io/webappsec-subresource-integrity/#agility
+     */
+    private async isIntegrityFormatValid(evt: FetchEnd): Promise<boolean> {
+        debug('Is integrity attribute valid?');
+        const { element, resource } = evt;
+        const integrity = element.getAttribute('integrity');
+        const integrityRegExp = /^sha(256|384|512)-/;
```

Comment:
What if I have multiple `integrity` attributes?

Reply:
I've added another test. Only the first instance is accessible via `getAttribute`. Additionally, it looks like Chrome tries to download the file twice if the first hash is invalid, but only the first attribute is accessible. I've left the code of the test commented with an explanation.